packages/subproviders/src/subproviders/nonce_tracker.ts

```diff
@@ -0,0 +1,97 @@
+import * as _ from 'lodash';
+
+import EthereumTx = require('ethereumjs-tx');
+import ethUtil = require('ethereumjs-util');
+import providerEngineUtils = require('web3-provider-engine/util/rpc-cache-utils');
+
+import {
+    BlockParamLiteral,
+    ErrorCallback,
+    JSONRPCPayload,
+    NonceSubproviderErrors,
+    OptionalNextCallback,
+} from '../types';
+
+import { Subprovider } from './subprovider';
+
+const NONCE_TOO_LOW_ERROR_MESSAGE = 'Transaction nonce is too low';
```

Comment:
Yeah let me quickly think this over, the reason I didn't move it is because its not an Error created by the nonce tracker, just a downstream error we'd like to handle.